theano/gof/opt.py

```diff
@@ -642,6 +642,8 @@ def process_node(self, fgraph, node):
                 continue
             if len(node.inputs) != len(candidate.inputs):
                 continue
+            if node.op != candidate.op:
+                continue
```

Comment:
TODO: remove that from this PR. It must be speed tested separatly. Not sure this is good to do. comparing scan op is time consuming.